src/Evolution/Systems/RelativisticEuler/Valencia/BoundaryCorrections/Rusanov.cpp

```diff
@@ -0,0 +1,250 @@
+
+#include <memory>
+#include <optional>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/RelativisticEuler/Valencia/Characteristics.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Formulation.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/NormalDotFlux.hpp"
+#include "PointwiseFunctions/Hydro/SoundSpeedSquared.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+/// \cond
+namespace RelativisticEuler::Valencia::BoundaryCorrections {
+template <size_t Dim>
+Rusanov<Dim>::Rusanov(CkMigrateMessage* msg) noexcept
+    : BoundaryCorrection<Dim>(msg) {}
+
+template <size_t Dim>
+std::unique_ptr<BoundaryCorrection<Dim>> Rusanov<Dim>::get_clone()
+    const noexcept {
+  return std::make_unique<Rusanov>(*this);
+}
+
+template <size_t Dim>
+void Rusanov<Dim>::pup(PUP::er& p) {
+  BoundaryCorrection<Dim>::pup(p);
+}
+
+template <size_t Dim>
+template <size_t ThermodynamicDim>
+double Rusanov<Dim>::dg_package_data(
+    const gsl::not_null<Scalar<DataVector>*> packaged_tilde_d,
+    const gsl::not_null<Scalar<DataVector>*> packaged_tilde_tau,
+    const gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*>
+        packaged_tilde_s,
+    const gsl::not_null<Scalar<DataVector>*> packaged_normal_dot_flux_tilde_d,
+    const gsl::not_null<Scalar<DataVector>*> packaged_normal_dot_flux_tilde_tau,
+    const gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*>
+        packaged_normal_dot_flux_tilde_s,
+    const gsl::not_null<Scalar<DataVector>*> packaged_abs_char_speed,
+
+    const Scalar<DataVector>& tilde_d, const Scalar<DataVector>& tilde_tau,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& tilde_s,
+
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& flux_tilde_d,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& flux_tilde_tau,
+    const tnsr::Ij<DataVector, Dim, Frame::Inertial>& flux_tilde_s,
+
+    const Scalar<DataVector>& lapse,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& shift,
+    const tnsr::ii<DataVector, Dim, Frame::Inertial>& spatial_metric,
+    const Scalar<DataVector>& rest_mass_density,
+    const Scalar<DataVector>& specific_internal_energy,
+    const Scalar<DataVector>& specific_enthalpy,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& spatial_velocity,
+
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& normal_covector,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& /*normal_vector*/,
+    const std::optional<tnsr::I<DataVector, Dim, Frame::Inertial>>&
+    /*mesh_velocity*/,
+    const std::optional<Scalar<DataVector>>& normal_dot_mesh_velocity,
+    const EquationsOfState::EquationOfState<true, ThermodynamicDim>&
+        equation_of_state) const noexcept {
+  {
+    // First compute v^2 and cs^2
+    Scalar<DataVector>& spatial_velocity_squared =
```

Comment:
(optional) `auto&`, and would be helpful to add a comment that the two return variables are used as memory buffer